pkg/elementalConfig/config.go

```diff
@@ -539,62 +544,31 @@ func NewBuildConfig(opts ...GenericOptions) *v1.BuildConfig {
 	return b
 }
 
-func ReadConfigRun(configDir string) (*v1.RunConfig, error) {
+// ReadConfigRunFromCloudConfig reads the configuration directly from a given cloud config string
+func ReadConfigRunFromCloudConfig(cc string) (*v1.RunConfig, error) {
 	cfg := NewRunConfig(WithLogger(v1.NewLogger()), WithOCIImageExtractor())
+	var err error
 
 	configLogger(cfg.Logger, cfg.Fs)
-
-	// TODO: is this really needed? It feels quite wrong, shouldn't it be loaded
-	// as regular environment variables?
-	// IMHO loading os-release as env variables should be sufficient here
-	cfgDefault := []string{"/etc/os-release"}
-	for _, c := range cfgDefault {
-		if exists, _ := utils.Exists(cfg.Fs, c); exists {
-			viper.SetConfigFile(c)
-			viper.SetConfigType("env")
-			cobra.CheckErr(viper.MergeInConfig())
-		}
-	}
-
-	// merge yaml config files on top of default runconfig
-	if exists, _ := utils.Exists(cfg.Fs, configDir); exists {
-		viper.AddConfigPath(configDir)
-		viper.SetConfigType("yaml")
-		viper.SetConfigName("config")
-		// If a config file is found, read it in.
-		err := viper.MergeInConfig()
-		if err != nil {
-			cfg.Logger.Warnf("error merging config files: %s", err)
-		}
-	}
-
-	// Load extra config files on configdir/config.d/ so we can override config values
-	cfgExtra := fmt.Sprintf("%s/config.d/", strings.TrimSuffix(configDir, "/"))
-	if exists, _ := utils.Exists(cfg.Fs, cfgExtra); exists {
-		viper.AddConfigPath(cfgExtra)
-		_ = filepath.WalkDir(cfgExtra, func(path string, d fs.DirEntry, err error) error {
-			if !d.IsDir() && filepath.Ext(d.Name()) == ".yaml" {
-				viper.SetConfigType("yaml")
-				viper.SetConfigName(strings.TrimSuffix(d.Name(), ".yaml"))
-				cobra.CheckErr(viper.MergeInConfig())
-			}
-			return nil
-		})
-	}
-
-	// unmarshal all the vars into the RunConfig object
-	err := viper.Unmarshal(cfg, setDecoder, decodeHook)
+	err = yaml.Unmarshal([]byte(cc), &cfg)
 	if err != nil {
-		cfg.Logger.Warnf("error unmarshalling RunConfig: %s", err)
+		return nil, err
 	}
-
+	// Store the full cloud-config in here so we can reuse it afterwards
+	cfg.FullCloudConfig = cc
 	err = cfg.Sanitize()
 	cfg.Logger.Debugf("Full config loaded: %s", litter.Sdump(cfg))
 	return cfg, err
 }
 
-func ReadInstallSpec(r *v1.RunConfig) (*v1.InstallSpec, error) {
+// ReadInstallSpecFromCloudConfig generates an installation spec from the raw cloud-config data
+func ReadInstallSpecFromCloudConfig(r *v1.RunConfig) (*v1.InstallSpec, error) {
```

Comment:
To be honest, this should be extracted into its own function and receive a runconfig + the key so we dont have duplication. 

Could also merge the config+spec into one function and return both, so get the cloud-config+action (upgrade,install,reset) string and return the config and the spec directly instead of doing it in 2 different places

Reply:
ah yes, this was done already, marking it as resolved!